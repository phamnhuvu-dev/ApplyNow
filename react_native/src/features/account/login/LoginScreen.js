"use strict";

import React, {Component} from "react";
import {AsyncStorage, Text, TouchableOpacity, View} from "react-native";
import {mainColor, whiteBlur50Color} from "../../../statics/Colors";
import WhiteTextInput from "../../../components/white-text-input";
import {BlueButton, WhiteButton} from "../../../components/button";
import {login} from "../../../modules/user-api";
import Routes from "../../../statics/Routes";

export default class LoginScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
    };
  }

  _email = (text) => this.state.email = text;
  _password = (text) => this.state.password = text;

  _goMain = () => this.props.navigation.replace(Routes.Main);
  _goSignUp = () => this.props.navigation.navigate(Routes.SignUp);
  _goForgotPassword = () => this.props.navigation.navigate(Routes.ForgotPassword);

  _login = () => {
    this._goMain();
    // login(this.state.email, this.state.password)
    //   .then(response => {
    //     if (response.data.login === "success") {
    //       this._storeData(response.data)
    //         .then(this._goMain)
    //         .catch(error => {
    //           console.log(error);
    //         });
    //       // this.props.dispatch(syncUser(response.data));
    //     }
    //   })
    //   .catch(error => {
    //     console.log(error);
    //   });
  };

  _storeData = async (user) => {
    await AsyncStorage.setItem("login", "1");
    await AsyncStorage.setItem("user", JSON.stringify(user));
  };

  render() {
    return (
      <View style={{flex: 1, backgroundColor: mainColor, justifyContent: "center"}}>
        <View style={{backgroundColor: whiteBlur50Color, borderRadius: 5, margin: 20, padding: 36}}>
          <WhiteTextInput
            placeholder="Email"
            onChangeText={this._email}
          />
          <WhiteTextInput
            placeholder="Password"
            secure
            onChangeText={this._password}
            onSubmitEditing={this._login}
          />

          <WhiteButton onPress={this._login} style={{marginTop: 20}} title="LOG IN"/>
          <BlueButton onPress={this._goSignUp} style={{marginTop: 8}} title="SIGN UP"/>
          <TouchableOpacity onPress={this._goForgotPassword}>
            <Text style={{
              marginTop: 8,
              color: "white",
              fontWeight: "bold",
              alignSelf: "center",
              textDecorationLine: "underline"
            }}>Forgot password?</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}