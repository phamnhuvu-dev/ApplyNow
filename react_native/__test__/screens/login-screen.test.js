'use strict';

import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import LoginScreen from "../../src/features/account/login-screen";

it('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});