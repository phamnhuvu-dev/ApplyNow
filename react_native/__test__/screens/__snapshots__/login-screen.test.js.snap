// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F06767",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff50",
        "borderRadius": 5,
        "margin": 20,
        "padding": 36,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff99"
      secureTextEntry={false}
      selectionColor="white"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "white",
          "color": "white",
          "fontSize": 16,
          "margin": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 4,
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#ffffff99"
      secureTextEntry={true}
      selectionColor="white"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "white",
          "color": "white",
          "fontSize": 16,
          "margin": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 4,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F06767",
            "fontWeight": "bold",
          }
        }
      >
        LOG IN
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#157FFB",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 8,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "bold",
            "marginTop": 8,
            "textDecorationLine": "underline",
          }
        }
      >
        Forgot password?
      </Text>
    </View>
  </View>
</View>
`;
